### PR TITLE
fix: add oracle manipulation

### PR DESCRIPTION
**Pull request by Tristan22400**

### CODE REVIEW

File: src/oracle-manipulation/oracle.sol [hunk 1]
```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract ERC20 {
```
[the remark this answers is not in the document]
Comment (makcandrov) [reply src/oracle-manipulation/oracle.sol]: Why Solmate over OZ? imo OZ would be more adapted as it is better known, and easier to read

[the remark this answers is not in the document]
Comment (makcandrov) [reply src/oracle-manipulation/oracle.sol]: Also `ERC20` from OZ is not marked as abstract, so we would no more need `ERC20Real`

[the remark this answers is not in the document]
Comment (MerlinEgalite) [reply src/oracle-manipulation/oracle.sol]: You can just import `ERC20Mock` from the `test` folder. Tbh i don't care. Juste the more we reuse code the faster

**Review by makcandrov — CHANGES_REQUESTED** [review]

File: src/oracle-manipulation/oracle.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tristan22400) [reply src/oracle-manipulation/oracle.sol]: In both case, we need to deploy a new contract to mint some tokens that can be added to the Linear Pool. So It is fine like that in my opinion.

**Comment by Tristan22400** [comment]
I proposed to close this PR to just keep the ones you open @makcandrov.

**Comment by makcandrov** [comment]
yep closing it in favor of #15

**Pull request closed without merging**